receiver/prometheusexecreceiver/subprocessmanager/manager.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subprocessmanager
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"math"
+	"math/rand"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/kballard/go-shellquote"
+	"go.uber.org/zap"
+)
+
+// Process struct holds all the info needed to instantiate a subprocess
+type Process struct {
+	Command string
+	Port    int
```

Comment:
Is `Port` part of some yet unimplemented feature?  I can't see any uses of it.  Is it supposed to be used to construct an arg to the subprocess telling it what port to run on?

Reply:
Yeah! This PR is one part of the bigger picture receiver `prometheus_exec` and `port` is important! thanks!

Reply:
EDIT: removing since this package is being moved to the Core repo, and I want to make it usable by all